app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -28,12 +28,16 @@ fi
 register_host() {
   curl --request POST <%= foreman_server_url %>/register \
        <%= headers.join(' ') %> \
-       -d '{ "host": { "name": "'$(hostname --fqdn)'", "build": "false", "managed": "false"
-       <%= ", \"organization_id\": \"#{@organization.id}\"" if @organization -%>
-       <%= ", \"location_id\": \"#{@location.id}\"" if @location -%>
-       <%= ", \"hostgroup_id\": \"#{@hostgroup.id}\"" if @hostgroup -%>
-       <%= ", \"operatingsystem_id\": \"#{@operatingsystem.id}\"" if @operatingsystem -%>
-       }}'
+       --data @- << EOF
+host[name]=$(hostname --fqdn)
+&host[build]=false
+&host[managed]=false
+<%= "&host[organization_id]=#{@organization.id}" if @organization -%>
+<%= "&host[location_id]=#{@location.id}" if @location -%>
+<%= "&host[hostgroup_id]=#{@hostgroup.id}" if @hostgroup -%>
+<%= "&host[operatingsystem_id]=#{@operatingsystem.id}" if @operatingsystem -%>
```

Comment:
You can repeat `--data` multiple times. No need to use `&` to concat.

Reply:
Can we keep it as it is? Repeating `--data "host[key]=value" \` for each parameter doesn't look much better IMO.

Reply:
I would write it as:
```suggestion
       <%= "--data 'host[organization_id]=#{@organization.id}' \\" if @organization -%>
       <%= "--data 'host[location_id]=#{@location.id}' \\" if @location -%>
       <%= "--data 'host[hostgroup_id]=#{@hostgroup.id}' \\" if @hostgroup -%>
       <%= "--data 'host[operatingsystem_id]=#{@operatingsystem.id}' \\" if @operatingsystem -%>
       --data "host[name]=$(hostname --fqdn)" \
       --data "host[build]=false" \
       --data "host[managed]=false"
```

I guess this is a style preference, but as a user this feels more natural to how I would pass it in. I'll leave it up to you and other reviewers what's the preferred style.

Reply:
I don't like multilines ERB and white space formatting shouldn't be done outside of erb tag for sake of code indentation. I'd personally use the `&` form since as clearer from the first look how to add more lines without understanding what the resulting command is. Leaving room for others to comment, to make it as easy as possible, :+1: on this comment means `&`, :-1: means `--data`.

Currently it's 2x :+1:  and 1x :-1:

Reply:
I want to be a special :unicorn: so I'll say the best would be shorhad of `--data` (`-d`).
But otherwise without `&` it's more human readable  and that's the focus here IMO, so :-1: from me.
No strong opinion here at all.

Reply:
> I want to be a special unicorn so I'll say the best would be shorhad of `--data` (`-d`).

In scripts this is a bad practice. If you're not familiar with the shorthand, you don't know what it is. Long options are self-documenting.

Reply:
perfect, I think it's 2:2 and no one with strong opinion :-) https://www.youtube.com/watch?v=CxK_nA2iVXw, I guess it's your call @stejskalleos

Reply:
or to make this easier, changing my vote, please use `--data` as suggested, it's better to make a decision than make no decision :-)